docs(policies): rework intro page and add concepts (#1956)
kumahq/kuma-website: app/_src/introduction/concepts.md

---
title: Concepts
---

In this page we will introduce concepts that are core to understanding Kuma.

> ## Control-Plane

Reviewer: There's no reason to use a hyphen here

The control-plane is the central management layer of {{ site.product_name }}. It is responsible for configuring and managing the behavior of the data plane,
which handles the actual traffic between services.

> ## Data-Plane Proxy / Sidecar

Reviewer 2: so is `Data-Plane Proxy` official way now of spelling it now? We have multiple kind of it thought of it.

We could use this concepts page as glossary for how to properly name stuff thorough our docs

Reviewer 2: also in policy introduction we use a lot just `proxy` should we include it in list: Data-Plane Proxy / Sidecar / Proxy?

Reviewer: We should add "control-plane" and "control-plane" to Vale to disallow them

Author: 100% I never remember what we should do so I'd love something in Vale.

So what is it? Data plane? Control Plane? or Data-plane and Control-plane?

The data-plane handles traffic between services.

> It connects to the control-plane which computes a configuration specific to it.

Reviewer 2: so should it be `control plane` as in first entry on this page or `control-plane` as here?

> The Data-Plane proxy or Sidecar is the instance of Envoy running alongside the data-plane which will send and receive traffic from the rest of the service mesh.

Reviewer: or capitalize these words

which handles the actual traffic between services.

## Data Plane

### Data Plane Proxy / Sidecar

The data plane handles traffic between services.
It connects to the control plane which computes a configuration specific to it.
The data plane proxy or sidecar is the instance of Envoy running alongside the application which will send and receive traffic from the rest of the service mesh.

{% mermaid %}
-------------------------------------
title: Representation of a Dataplane
-------------------------------------
flowchart LR

clients@{ shape: processes, label: "clients" }
servers@{ shape: processes, label: "servers"}

subgraph Data plane
   app
   subgraph data plane proxy/Envoy
   inbounds
   outbounds
   end
end

inbounds -.local traffic.-> app
app -.local traffic.-> outbounds

clients --> inbounds
outbounds --> servers
{% endmermaid %}

### Inbound

An inbound is the part of the data plane proxy which receives traffic for a specific port.
inbounds are usually grouped between different data planes and form a service.

### Outbound

> An outbound is the part 

Reviewer: Jut adding a comment so we don't forget

## Resource

A resource is an object or entity that can be created, managed, and interacted with in {{ site.product_name }}.

> Resources are the building blocks that define the behavior and state of your service-mesh.

Reviewer: ```suggestion
Resources are the building blocks that define the behavior and state of your service mesh.
```

Each resource is defined as a type of API object that has a specific purpose and is represented by its state and configuration.

A resource is most often expressed as yaml and can have 2 formats:

- `Kubernetes` when the backing control plane runs on Kubernetes. In this case {{ site.product_name }} resources are defined as Kubernetes Custom Resource Definitions.
- `Universal` in other cases or when accessing resources through {{ site.product_name}}'s REST api.

Resources are most commonly represented in yaml format.

## Policy

> Policies are a specific type of Resources which controls the behaviour, security and communication of applications running inside your service mesh.

Reviewer: ```suggestion
Policies are a specific type of resource that controls the behaviour and communication of applications running inside your service mesh.
```

just cause security is in the next line
